contracts/multiply/GoerliDummyExchange.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity >=0.7.6;
+import "../interfaces/IERC20.sol";
+import "../utils/SafeMath.sol";
+import "../utils/SafeERC20.sol";
+import "hardhat/console.sol";
+
+contract GoerliDummyExchange {
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20;
+
+  address DAI_ADDRESS = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+  mapping(address => bool) public WHITELISTED_CALLERS;
+
+  uint8 slippage;
+
+  uint8 public fee = 0;
+  uint256 public feeBase = 10000;
+
+  address public feeBeneficiaryAddress = 0x70997970C51812dc3A010C7d01b50e0d17dc79C8; // second HH address
+
+  event AssetSwap(
+    address indexed assetIn,
+    address indexed assetOut,
+    uint256 amountIn,
+    uint256 amountOut
+  );
+  
+  modifier onlyAuthorized() {
+    require(WHITELISTED_CALLERS[msg.sender], "Exchange / Unauthorized Caller.");
+    _;
+  }
+
+  constructor(address _beneficiary, uint8 _fee, uint8 _slippage, address _dai, address authorisedCaller){
+    feeBeneficiaryAddress = _beneficiary;
+    fee = _fee;
+    slippage = _slippage;
+    DAI_ADDRESS = _dai;
+    WHITELISTED_CALLERS[authorisedCaller] = true;
+  }
+  
+  event FeePaid(address indexed beneficiary, uint256 amount);
+  event SlippageSaved(uint256 minimumPossible, uint256 actualAmount);
+
+  function _transferIn(
+    address from,
+    address asset,
+    uint256 amount
+  ) internal {
+    console.log("TRANSFER IN", asset, amount);
+    require(
+      IERC20(asset).allowance(from, address(this)) >= amount,
+      "Exchange / Not enought allowance"
+    );
+    require(
+      IERC20(asset).balanceOf(from)>=amount,
+      "Exchange / Could not swap"
+    );
+    IERC20(asset).transferFrom(from, address(this), amount);
+  }
+
+  function _transferOut(
+    address asset,
+    address to,
+    uint256 amount
+  ) internal {
+    IERC20(asset).safeTransfer(to, amount);
+    emit SlippageSaved(amount, amount);
+    console.log("TRANSFER OUT", asset, amount);
+  }
+
+  function _collectFee(address asset, uint256 fromAmount) public returns (uint256) {
+    uint256 feeToTransfer = fromAmount.mul(fee).div(feeBase);
+    IERC20(asset).transferFrom(address(this), feeBeneficiaryAddress, feeToTransfer);
+    emit FeePaid(feeBeneficiaryAddress, feeToTransfer);
+    return fromAmount.sub(feeToTransfer);
+  }
+
+  // uses the same interface as default Exchange contract
+  function swapDaiForToken(
```

Comment:
So interfaces are reused but there's no syntax to explicitly show that this is the same type of an interface in default and dummy exchange ?